### DISCUSSION TITLE
Question about data interpretation

### DISCUSSION THREAD

**Discussion by sneddonucsf** [theislab/cellrank#893] [Biological applications] [answered]
Hello, 

Thank you for the great package! 
I am trying to use CellRank on my dataset of developing human fetal endocrine cells. Other pseudotime methods (Slingshot) has constructed the lineages as such: 
<img width="590" alt="Screen Shot 2022-06-10 at 12 26 35 PM" src="https://user-images.githubusercontent.com/12053091/173136397-eca56f40-3c5f-4378-919a-293d3a7d2cdf.png">

 
So 4 terminally differentiated hormone+ cells, and 4 progenitor populations (clusters 2.0.0 -> 2.1.1) with varying degrees of differentiation potency. Because RNA Velocity and CytoScape do not work for my data, I am attempting to use the pseudotime kernel on my data with diffusion pseudotime from Scanpy. First, I've converted my Seurat object to an Adata object and then calculated psedotime with the following code: 

``` 
adata.uns['iroot'] = np.flatnonzero(adata.obs['annotation']  == 'cEP')[0]
sc.tl.dpt(adata, n_branchings = 0)
```
This gives me the following pseudotime measurements: 
<img width="412" alt="Screen Shot 2022-06-10 at 1 06 11 PM" src="https://user-images.githubusercontent.com/12053091/173142707-8211f78a-38be-4361-9b96-72c629274a6b.png">
As you can see, the pseudotime values for the beta cells is low compared to the rest of the hormone+ cells. For reference, PAGA looks as the following: 
<img width="793" alt="Screen Shot 2022-06-10 at 1 04 11 PM" src="https://user-images.githubusercontent.com/12053091/173142441-5702b248-1ded-48a4-b2c4-aa46c98c8cad.png">
I've then followed through the rest of the pseudotime kernel/estimators:
```
pk = cr.tl.kernels.PseudotimeKernel(adata)
pk.compute_transition_matrix(threshold_scheme='soft')
scv.pl.velocity_embedding_stream(
    adata, color="annotation", vkey= "T_fwd", basis="umap", legend_loc="right", add_outline = True)
```
<img width="546" alt="Screen Shot 2022-06-10 at 12 47 50 PM" src="https://user-images.githubusercontent.com/12053091/173139676-2d45a4c9-77b8-48fc-8979-add34108c9e8.png">
This generally agrees with our Slingshot lineage reconstruction

```
from cellrank.tl.estimators import GPCCA
g_fwd = GPCCA(pk)
g_fwd.compute_schur(n_components=20)
g_fwd.plot_spectrum(real_only=True)
```
This gives me a eigenvalue of 3, but calculating the macrostates with this value doesn't quite capture what I think are true macrostates: 
<img width="396" alt="Screen Shot 2022-06-10 at 12 34 02 PM" src="https://user-images.githubusercontent.com/12053091/173137454-31ba4a30-277e-4b8d-b7c1-25b4a40eaf30.png">
So I calculated 5 macrostates, essentially the beginning of my trajectories (cEP) and the terminally differentiated cells types: 
<img width="407" alt="Screen Shot 2022-06-10 at 12 34 50 PM" src="https://user-images.githubusercontent.com/12053091/173137652-fcae2372-1392-416b-88ef-17d62a6b3843.png">

```
g_fwd.set_terminal_states_from_macrostates(["Alpha", "Beta", "Epsilon", "Delta"])
g_fwd.plot_coarse_T()
```
The coarse plot looks:  
<img width="653" alt="Screen Shot 2022-06-10 at 2 51 58 PM" src="https://user-images.githubusercontent.com/12053091/173155805-f8451826-7a05-41bf-9deb-65eabd16abcb.png">
with the cEP and Beta clusters being denoted as early states. I would not expect beta cells to be denoted as such. When I calculate the absorption probabilities, I get: 
```
g_fwd.plot_terminal_states(same_plot=True)
g_fwd.compute_absorption_probabilities()
g_fwd.compute_absorption_probabilities(time_to_absorption="all")
g_fwd.plot_absorption_probabilities()
```
<img width="345" alt="Screen Shot 2022-06-10 at 2 50 00 PM" src="https://user-images.githubusercontent.com/12053091/173155629-c2a47c89-f712-419c-93bd-490169bf0d96.png">
<img width="343" alt="Screen Shot 2022-06-10 at 2 50 06 PM" src="https://user-images.githubusercontent.com/12053091/173155634-f09f0804-2a9b-4e49-9c57-9f31dc8e77df.png">
<img width="390" alt="Screen Shot 2022-06-10 at 2 50 12 PM" src="https://user-images.githubusercontent.com/12053091/173155640-90a5c396-9300-4d37-832d-6c90d280694c.png">

I have the following questions: 
1.) Can the pseudotime kernel/estimator results be used for other downstream applications such as directed PAGA, circular projection, plotting lineages, etc.? 
2.) In the absorption probabilities plot, is the interpretation that cells that are colored as the terminal state cluster are "fated" to become that cell type? 
3.) The absorption probabilities for Beta cells is low as compared to the other terminal states - I'm guessing that this is due to the lower pseudotime values as compared to the other hormone+ cells. Would you recommend trying to get better pseudotime reconstruction and redoing the kernel/estimators? How can I infer lineages? 
Thank you very much!

**Comment by Marius1311** [comment, marked as answer]
Hi @sneddonucsf, thanks for using Cellrank. First question from my side; if you have a slingshot pseudotime and it works well on your data, then why don't you just use that for the pseudotime kernel? You may use *any* pseudotime method that works for your data!

With respect to your specific questions...
1. yes
2. yes; you can view this more specifically by visualizing one trajectory per plot, check the API (I think `same_plot=False`)
3. yes, just use your slingshot pseudotime if you're happy with it; that should improve your fate probabilities.

**Reply by sneddonucsf** [reply]
Hi @Marius1311, thank you for the reply! I've tweaked my pseudotime calculation and now the data looks much better: 
<img width="383" alt="Screen Shot 2022-06-14 at 12 19 26 PM" src="https://user-images.githubusercontent.com/12053091/173671677-d0a69d42-43ec-40cc-a8c6-3b2c9a829bc1.png">
<img width="346" alt="Screen Shot 2022-06-14 at 12 19 59 PM" src="https://user-images.githubusercontent.com/12053091/173671684-79a02f9d-dd00-47fa-876e-7506bc57baf6.png">
<img width="1000" alt="Screen Shot 2022-06-14 at 12 20 18 PM" src="https://user-images.githubusercontent.com/12053091/173671689-0665fb2e-3d16-41b3-8b21-6b311c382db2.png">

Sorry, I have some more questions: 
1.) For the macrostate computation, how many "states" would one expect in biologically? I know that you can calculate this and look at the eigenvalues, but in my case the value was 3, which would not give me the expected terminal states. Is it valid to increase this number based on biological interpretation? And would we expect the most initial progenitor (for example, the Ngn3_Low population in the mouse pancreas tutorial) to be a macrostate? Just wondering if I should increase the value to capture all of my terminal states as well as the most initial progenitor. 
2.) For directed PAGA with pseudotime data, is it required to have the initial_state? I know that in the directed PAGA tutorial uses ```cellrank.tl.initial_states()```, but I believe this is calculated with velocity information? 

Thanks again!

**Reply by Marius1311** [reply]
Hi! 
Re 1: I would say it depends what your primary interest is. If this is a well studied system and you don't expect to find novel initial/terminal states (which I believe is the case), then I would argue it's fine to set the initial and terminal states in a semi-automatic fashion by computing macrostates and manually declaring some of them to be initial/terminal. This enables you to compute fate probabilities which in your case is probably most relevant - it allows you to zoom in on regulation, which genes may be implicated in Alpha vs. Beta differentiation for example?
Re 2: I'm not exactly sure how to do this and I think I've never tried this, please have a look trough closed issues whether anyone tried this before and if not feel free to open a new one specifically for this.

**Reply by sneddonucsf** [reply]
@Marius1311 That makes sense, in this case we know what to expect for the initial and terminal states, it's more which intermediate states lead to which terminal (so, constructing the lineage). Thank you very much for your help!
